deps/rabbitmq_prometheus/src/collectors/prometheus_rabbitmq_core_metrics_collector.erl

```diff
@@ -579,6 +603,42 @@ get_data(Table, false, VHostsFilter) when Table == channel_exchange_metrics;
         _ ->
             [Result]
     end;
+get_data(exchange_metrics = Table, true, VHostsFilter) when is_map(VHostsFilter)->
+    ets:foldl(fun
+        ({#resource{kind = exchange, virtual_host = VHost}, _, _, _, _, _} = Row, Acc) when
+            map_get(VHost, VHostsFilter)
+        ->
+            [Row | Acc];
+        (_Row, Acc) ->
+            Acc
+    end, [], Table);
+get_data(exchange_metrics, true, _VhostsFilter) ->
+    [];
```

Comment:
this is the opposite of the intended behaviour. If `VHostsFilter` is undefined (`false`) then metrics from all vhosts should be included
After second thought this clauses is not needed as the last clause of `get_data` which just calls ` ets:tab2list(Table).` will cover what is needed